src/fmeval/eval_algorithms/context_quality.py

```diff
@@ -0,0 +1,291 @@
+        return record
+
+    def _get_score(self, model_input: str, target_output: str, retrieved_context: str) -> float:
+        """Computes the context precision score by composing a prompt on each (context chunk, model input,
+        target output) triple and doing inference on the judge model to get a verdict of 0 or 1, then
+        calculating the score using the judge model verdicts.
+
+        For multiple target contexts, we do a "logical or" operation on the model verdicts, so if the
+        model verdict for a given context chunk is 1 for any target output, the model verdict for that
+        context chunk will be 1.
+
+        :param model_input: The input that is composed with a prompt template and passed to the model.
+        :param target_output: The referenced "ground truth" answer.
+        :param retrieved_context: The context retrieved from the RAG system.
+        :return: the context precision score.
+        """
+        target_outputs = target_output.split(self.target_output_delimiter)
+        context_chunks = retrieved_context.split(self.retrieved_context_delimiter)
+        results = []
+        for context_chunk in context_chunks:
+            model_verdict = 0
+            for target_output in target_outputs:
+                prompt = self.context_precision_prompt_composer.compose(
+                    model_input, {"target_output": target_output, "retrieved_context": context_chunk}
+                )
+                model_output = self.judge_model.predict(prompt)[0]
+                model_verdict = ContextPrecisionScore._parse_model_output(model_output)
+                if model_verdict:
+                    break
+            results.append(model_verdict)
+        score = ContextPrecisionScore._compute_metric(results)
+        return score
+
+    @staticmethod
+    def _parse_model_output(model_output: str) -> int:
+        """Parses the model output to a binary score of 0 or 1
+
+        :param model_output: The output of the judge model.
+        :return: Score of 0 or 1.
+        """
+        response_words = model_output.split(" ")
+        predicted_labels = [
+            word.lower().strip() for word in response_words if word.lower().strip() in BINARY_SCORE_VALUES
+        ]
+        string_label = predicted_labels[0] if predicted_labels else "0"
+        return int(string_label)
+
+    @staticmethod
+    def _compute_metric(judge_model_verdicts: List[int]) -> float:
+        """Computes the context precision score for one record given judge model verdicts
+
+        :param judge_model_verdicts: A list of judge model verdicts of 0 or 1.
+        :return: the context precision score for the record.
+        """
+        response_list = [1 if ver else 0 for ver in judge_model_verdicts]
+        denominator = sum(response_list) + 1e-10
+        numerator = sum([(sum(response_list[: i + 1]) / (i + 1)) * response_list[i] for i in range(len(response_list))])
+        score = numerator / denominator
```

Comment:
We should probably warn the user if `response_list` is empty, indicating that the judge model did not produce intended output.